TS.fsx

```diff
@@ -49,22 +50,26 @@ let rec DomTypeToTsType (objDomType: string) =
                 allCallbackFuncs.ContainsKey objDomType ||
                 allDictionariesMap.ContainsKey objDomType then
                 objDomType
+            // Name of a type alias. Just return itself
+            elif typeDefSet.Contains objDomType then objDomType
             // Enum types are all treated as string
             elif allEnumsMap.ContainsKey objDomType then "string"
             // Union types
-            elif (objDomType.Contains(" or ")) then
+            elif objDomType.Contains(" or ") then
                 let allTypes = objDomType.Trim('(', ')').Split([|" or "|], StringSplitOptions.None)
-                                |> Array.map DomTypeToTsType
+                                |> Array.map (fun t -> DomTypeToTsType (t.Trim('?', ' ')))
                 if Seq.contains "any" allTypes then "any" else String.concat " | " allTypes
             else
                 // Check if is array type, which looks like "sequence<DOMString>"
-                let genericMatch = Regex.Match(objDomType, @"^(\w+)<(\w+)>$")
+                let unescaped = objDomType.Replace("&lt;", "<").Replace("&gt;", ">")
+                let genericMatch = Regex.Match(unescaped, @"^(\w+)<(\w+)>$")
```

Comment:
The nested case should be taken care of, as the implementation is recursive. In line 69 the `paramName` is obtained via another run of `DomTypeToTsType` using the matched part, so even if it has a type argument, it will be recognized.

As for the case with multiple type arguments, the logic would become much more complex, because it is non-trivial to tell whether ',' delimits two types or is just inside a type argument. Up to now I haven't seen the case appeared yet in the spec, it might be worth another PR to address the issue.